src/diffusers/models/resnet.py

```diff
@@ -446,10 +446,15 @@ def upfirdn2d_native(input, kernel, up=1, down=1, pad=(0, 0)):
     kernel_h, kernel_w = kernel.shape
 
     out = input.view(-1, in_h, 1, in_w, 1, minor)
+    using_mps = out.device.type == "mps"
+    if using_mps:
```

Comment:
this deserves a comment here IMO -> are some kernels not yet supported on `"mps"`?

Reply:
Yes, that causes a runtime error: https://github.com/pytorch/pytorch/issues/84535